[tuya] Exception, channel unknown (airpurifier)

Error message when adding a "Smartmi Air Purifier P1", PID: nedhywwbir5vbypd, Vers: 3.3 
It was added through the UI/discovery (no textual config).

## Current Behavior
Throws exception and thing can't initialize
`2026-01-09 16:08:44.759 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.tuya.internal.handler.TuyaDeviceHandler@19b14f4b': Channel type 'tuya:nedhywwbir5vbypd_filter_useH' is not known`

## Your Environment
OH5.1

Indeed, there's a schema missing message there. I disabled/enabled the "thing" for the device with the binding set to debug: 

`2026-01-11 18:02:41.948 [WARN ] [uya.internal.TuyaChannelTypeProvider] - No schema for product nedhywwbir5vbypd channel type filter_useh
2026-01-11 18:02:41.949 [ERROR] [nal.common.AbstractInvocationHandler] - An error occurred while calling method 'ThingHandler.initialize()' on 'org.openhab.binding.tuya.internal.handler.TuyaDeviceHandler@6ddd8030': Channel type 'tuya:nedhywwbir5vbypd_filter_useH' is not known
java.lang.IllegalArgumentException: Channel type 'tuya:nedhywwbir5vbypd_filter_useH' is not known
        at org.openhab.core.thing.internal.ThingHandlerCallbackImpl.createChannelBuilder(ThingHandlerCallbackImpl.java:222) ~[?:?]
        at org.openhab.binding.tuya.internal.handler.TuyaDeviceHandler.lambda$10(TuyaDeviceHandler.java:659) ~[?:?]
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
        at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939) ~[?:?]
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?]
        at org.openhab.binding.tuya.internal.handler.TuyaDeviceHandler.addChannels(TuyaDeviceHandler.java:663) ~[?:?]
        at org.openhab.binding.tuya.internal.handler.TuyaDeviceHandler.initialize(TuyaDeviceHandler.java:567) ~[?:?]
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:149) ~[?:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
        at java.lang.Thread.run(Thread.java:1583) [?:?]
2026-01-11 18:02:41.951 [ERROR] [core.thing.internal.ThingManagerImpl] - Exception occurred while initializing handler of thing 'tuya:tuyaDevice:airpurifier_fe_stue': Channel type 'tuya:nedhywwbir5vbypd_filter_useH' is not known
java.lang.IllegalArgumentException: Channel type 'tuya:nedhywwbir5vbypd_filter_useH' is not known
        at org.openhab.core.thing.internal.ThingHandlerCallbackImpl.createChannelBuilder(ThingHandlerCallbackImpl.java:222) ~[?:?]
        at org.openhab.binding.tuya.internal.handler.TuyaDeviceHandler.lambda$10(TuyaDeviceHandler.java:659) ~[?:?]
        at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
        at java.util.Iterator.forEachRemaining(Iterator.java:133) ~[?:?]
        at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1939) ~[?:?]
        at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
        at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
        at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
        at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
        at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?]
        at org.openhab.binding.tuya.internal.handler.TuyaDeviceHandler.addChannels(TuyaDeviceHandler.java:663) ~[?:?]
        at org.openhab.binding.tuya.internal.handler.TuyaDeviceHandler.initialize(TuyaDeviceHandler.java:567) ~[?:?]
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:580) ~[?:?]
        at org.openhab.core.internal.common.AbstractInvocationHandler.invokeDirect(AbstractInvocationHandler.java:149) ~[?:?]
        at org.openhab.core.internal.common.Invocation.call(Invocation.java:52) ~[?:?]
        at java.util.concurrent.FutureTask.run(FutureTask.java:317) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[?:?]
        at java.lang.Thread.run(Thread.java:1583) [?:?]`

--- a/bundles/org.openhab.binding.tuya/src/main/java/org/openhab/binding/tuya/internal/TuyaChannelTypeProvider.java
+++ b/bundles/org.openhab.binding.tuya/src/main/java/org/openhab/binding/tuya/internal/TuyaChannelTypeProvider.java
@@ -167,7 +167,7 @@ public class TuyaChannelTypeProvider implements ChannelTypeProvider {
         channelTypeId = channelTypeId.substring(i + 1);
 
         // Build with a channelTypeId of just the lower-cased DP identifier and set defaults for all text.
-        channelType = channelTypeFromSchema(channelTypeUID, productId, channelTypeId.toLowerCase());
+        channelType = channelTypeFromSchema(channelTypeUID, productId, channelTypeId);
         if (channelType != null) {
             // Localize that (e.g. using channel-type.tuya.cur_voltage.label = ...)
             channelType = localizationService.createLocalizedChannelType(bundle, channelType, locale);

